### PR TITLE
Master

### PR DESCRIPTION
**Pull request by ngnvnct**
[//]: # "Credit: https://github.com/neetcode-gh/leetcode/blob/main/.github/pull_request_template.md"
[//]: # "Pull Request Template"
[//]: # "Replace the placeholder values in the template below"

- **File(s) Submission**: problem_0198_house_robber.py
- **Programming Language(s)**: python
- **Submission URL**: https://leetcode.com/problems/house-robber/

### CODE REVIEW

File: python/dynamic programming/problem_0198_house_robber.py [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+Time complexity = O(n) tabulation dp. Only have to iterate through the array once
+Space = O(n) We have to make a completely separate dp array.
+"""
+
+class TestCase(unittest.TestCase):
+    def setUp(self):
+        self.solution = Solution()
+
+        self.testCase1 = [1,2,3,1]
```
Comment (cetsai) [inline python/dynamic programming/problem_0198_house_robber.py:27]: Move these variables into their respective testing functions— no need for them to be in the class scope

File: python/dynamic programming/problem_0198_house_robber.py [hunk 2]
```diff
@@ -0,0 +1,39 @@
+import unittest
+
+class Solution:
+    @staticmethod
+    def rob(nums):
```
Comment (cetsai) [inline python/dynamic programming/problem_0198_house_robber.py:5]: Consider renaming nums to something more meaningful.

**Review by cetsai — CHANGES_REQUESTED** [review]
Add a test case for the edge case you described as well as for len(nums) == 1

**Comment by cetsai** [comment]
Rename pr

File: python/dynamic programming/problem_0198_house_robber.py (hunk 2 above)
Comment (thuanle123) [reply python/dynamic programming/problem_0198_house_robber.py:5]: we don't need to rename Leetcode input parameters

class Solution:
    def rob(self, nums: List[int]) -> int:

File: python/dynamic programming/problem_0198_house_robber.py (hunk 1 above)
Comment (thuanle123) [reply python/dynamic programming/problem_0198_house_robber.py:27]: Tests can be numerous, and their set-up can be repetitive. Luckily, we can factor out set-up code by implementing a method called [setUp()](https://docs.python.org/3/library/unittest.html#unittest.TestCase.setUp), which the testing framework will automatically call for every single test we run:

https://docs.python.org/3/library/unittest.html#module-unittest

That is correct, you only need to keep self.solution = Solution() in the setUp method, the rest should go into their respective test folders

Comment (thuanle123) [reply python/dynamic programming/problem_0198_house_robber.py:27]: The setUp() method is called before every test function is executed. In this particular case, testCase1 is a common input for multiple test functions, so it makes sense to define it in the setUp() method to avoid repeating the same code in every test function.

By defining testCase1 in setUp(), it becomes an instance variable that is accessible to all test functions within the test class. This way, any changes made to testCase1 in one test function do not affect the value of testCase1 in other test functions, which can be useful for keeping test cases independent from each other.

never it's fine

**Comment by thuanle123** [comment]
https://leetcode.com/problems/house-robber/submissions/927246433/

like this

you want the submission URL to look like this btw (go into the submission tab and click on the Accepted Solution that you commited, and copy the URL
